### PR TITLE
fix: [CDS-75172]: disallow '.' (at first char) and '$' from Identifier and versionLabel also corrected

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
### Summary

Changes in the Regex pattern - 
 - regexIdentifier: disallowing `'.' (at first char) and '$'`
 - regexVersionLabel: disallowing `/ and &`
 
 So now allowed would be - 
 - regexIdentifier: must start with a letter or _ and can then be followed by alphanumerics or _
 - regexVersionLabel: must start with alphanumerics and should not contain whitespaces, / or &
 
 Reason required for parity between Yaml/BE validation and UI form validation.
 

#### Screenshots

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by srdjan-harness — APPROVED** [review]

File: src/modules/10-common/utils/StringUtils.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export const regexUsernameDisallowedChars = /^[^/:<>=()]+$/ // for username
 
 export const regexPositiveNumbers = /^[1-9]+[0-9]*$/
 
-export const regexIdentifier = /^[a-zA-Z_.][0-9a-zA-Z_$]*$/
+export const regexIdentifier = /^[a-zA-Z_][0-9a-zA-Z_]*$/
```
Comment (srdjan-harness) [inline src/modules/10-common/utils/StringUtils.ts:61]: This does not allow dot (.) at all. Is this correct as PR description contains (at first char)?

Comment (srdjan-harness) [reply src/modules/10-common/utils/StringUtils.ts:61]: it was allowed only at the begginig .Ignore my comment.

Comment (pankajsingh-harness) [reply src/modules/10-common/utils/StringUtils.ts:61]: @Sagar-Rajput7 is this regex the same as used in BE? We should have same validation in UI and BE

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/10-common/utils/StringUtils.ts (hunk 1 above)
Comment (Sagar-Rajput7) [reply src/modules/10-common/utils/StringUtils.ts:61]: yes

**Pull request merged**